File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentCronJobRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentCronJobModel;

public interface IAgentCronJobRepository {
    CompletableFuture<List<AgentCronJobModel>> listAsync(UUID agentId);
    CompletableFuture<List<AgentCronJobModel>> listAllEnabledAsync();
    CompletableFuture<AgentCronJobModel> getAsync(UUID id);
    CompletableFuture<AgentCronJobModel> createAsync(UUID agentId, String name, String expression, String prompt);
    CompletableFuture<Void> updateAsync(AgentCronJobModel model);
    CompletableFuture<Void> setEnabledAsync(UUID id, boolean enabled);
    CompletableFuture<Boolean> deleteAsync(UUID id);
}

File: apps/backend/src/Features/Agents/Domain/IAgentCronJobRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IAgentCronJobRepository
{
    Task<IReadOnlyList<AgentCronJobRecord>> ListAsync(Guid agentId, CancellationToken ct = default);
    Task<IReadOnlyList<AgentCronJobWithAgentRecord>> ListForOwnerAsync(Guid ownerId, CancellationToken ct = default);
    Task<IReadOnlyList<AgentCronJobRecord>> ListAllEnabledAsync(CancellationToken ct = default);
    Task<AgentCronJobRecord?> GetByAsync(AgentCronJobFilter filter, CancellationToken ct = default);
    Task<AgentCronJobWithAgentRecord?> GetForOwnerAsync(Guid id, Guid ownerId, CancellationToken ct = default);
    Task<AgentCronJobRecord> CreateAsync(Guid agentId, string name, string expression, string prompt, CancellationToken ct = default);
    Task UpdateAsync(AgentCronJobRecord record, CancellationToken ct = default);
    Task SetEnabledAsync(Guid id, bool enabled, CancellationToken ct = default);
    Task<bool> DeleteAsync(Guid id, CancellationToken ct = default);
}
